File: backend/models/Booking.py
# cortate/backend/models/Booking.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class Booking(BaseModel):
    id: Optional[str] = Field(None, alias="_id")
    barbero_id: str  # ID del barbero
    cliente_id: str  # ID del cliente
    fecha: str  # formato esperado: 'YYYY-MM-DD'
    hora: str  # formato esperado: 'HH:MM'
    servicio: str  # tipo de servicio: corte, barba, etc.
    tipo: str  # 'domicilio' o 'local'
    estado: str = "pendiente"  # otros posibles: confirmado, cancelado, finalizado
    creado_en: Optional[datetime] = Field(default_factory=datetime.utcnow)

    class Config:
        allow_population_by_field_name = True
        arbitrary_types_allowed = True
        json_encoders = {
            datetime: lambda v: v.isoformat(),
        }

File: backend/models/Booking.js
// cortate/backend/models/Booking.js

const mongoose = require('mongoose');

const BookingSchema = new mongoose.Schema({
	cliente: {
		type: mongoose.Schema.Types.ObjectId,
		ref: 'User',
		required: true
	},
	barbero: {
		type: mongoose.Schema.Types.ObjectId,
		ref: 'Barber',
		required: true
	},
	servicio: {
		type: String,
		enum: ['corteHombre', 'corteBarba', 'corte+barba', 'ninos', 'express', 'diseno', 'padreHijo'],
		required: true
	},
	monto: {
		type: Number,
		required: true
	},
	fechaReserva: {
		type: Date,
		required: true
	},
	horaReserva: {
		type: String,
		required: true
	},
	estado: {
		type: String,
		enum: ['pendiente', 'confirmado', 'realizado', 'cancelado', 'rechazado'],
		default: 'pendiente'
	},
	creadoEn: {
		type: Date,
		default: Date.now
	},
	confirmadoManual: {
		type: Boolean,
		default: false
	},
	canceladoPor: {
		type: String,
		enum: ['cliente', 'barbero', null],
		default: null
	},
	tipoAtencion: {
		type: String,
		enum: ['local', 'domicilio'],
		required: true
	},
	penalizacion: {
		type: Boolean,
		default: false
	}
});

module.exports = mongoose.model('Booking', BookingSchema);
